ts/features/idpay/initiative/details/screens/BeneficiaryDetailsScreen.tsx

```diff
@@ -0,0 +1,389 @@
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { RouteProp, useNavigation, useRoute } from "@react-navigation/native";
+import { sequenceS } from "fp-ts/lib/Apply";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import Placeholder from "rn-placeholder";
+import { InitiativeDTO } from "../../../../../../definitions/idpay/InitiativeDTO";
+import { InitiativeDetailDTO } from "../../../../../../definitions/idpay/InitiativeDetailDTO";
+import { ContentWrapper } from "../../../../../components/core/ContentWrapper";
+import { Icon } from "../../../../../components/core/icons";
+import { HSpacer, VSpacer } from "../../../../../components/core/spacer/Spacer";
+import { Body } from "../../../../../components/core/typography/Body";
+import { H3 } from "../../../../../components/core/typography/H3";
+import { H4 } from "../../../../../components/core/typography/H4";
+import { Link } from "../../../../../components/core/typography/Link";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
+import ButtonSolid from "../../../../../components/ui/ButtonSolid";
+import Markdown from "../../../../../components/ui/Markdown";
+import I18n from "../../../../../i18n";
+import {
+  AppParamsList,
+  IOStackNavigationProp
+} from "../../../../../navigation/params/AppParamsList";
+import { useIODispatch, useIOSelector } from "../../../../../store/hooks";
+import { format } from "../../../../../utils/dates";
+import { useIOBottomSheetModal } from "../../../../../utils/hooks/bottomSheet";
+import { useOnFirstRender } from "../../../../../utils/hooks/useOnFirstRender";
+import { formatNumberAmount } from "../../../../../utils/stringBuilder";
+import { openWebUrl } from "../../../../../utils/url";
+import { Table, TableItem } from "../../../common/components/Table";
+import { IDPayUnsubscriptionRoutes } from "../../../unsubscription/navigation/navigator";
+import { IDPayDetailsParamsList } from "../navigation";
+import {
+  idPayBeneficiaryDetailsSelector,
+  idpayInitiativeDetailsSelector
+} from "../store";
+import { idPayBeneficiaryDetailsGet } from "../store/actions";
+
+export type BeneficiaryDetailsScreenParams = {
+  initiativeId: string;
+  initiativeName?: string;
+};
+
+type BeneficiaryDetailsScreenRouteProps = RouteProp<
+  IDPayDetailsParamsList,
+  "IDPAY_DETAILS_BENEFICIARY"
+>;
+
+const formatNumberCurrency = (amount: number) =>
+  `${formatNumberAmount(amount, false)} €`;
+
```

Comment:
could this be added as a reusable utility func?

Reply:
In this case this function (and `formatDate` function) are wrappers that can be used inside `pipe`.  `formatNumberAmount` already is an exported utility function.